samples/17.multilingual-bot/src/main/java/com/microsoft/bot/sample/multilingual/MultiLingualBot.java

```diff
@@ -1,2 +1,173 @@
+            String lang = currentLang.equals(ENGLISH_ENGLISH) || currentLang.equals(SPANISH_ENGLISH)
+                ? ENGLISH_ENGLISH : ENGLISH_SPANISH;
+
+            // If the user requested a language change through the suggested actions with values "es" or "en",
+            // simply change the user's language preference in the user state.
+            // The translation middleware will catch this setting and translate both ways to the user's
+            // selected language.
+            // If Spanish was selected by the user, the reply below will actually be shown in spanish to the user.
+            Activity reply = MessageFactory.text(String.format("Your current language code is: %1$s", lang));
+            return languagePreference.set(turnContext, lang)
+                .thenCompose(task -> turnContext.sendActivity(reply))
+                .thenCompose(task -> userState.saveChanges(turnContext, false));
+        } else {
+            // Show the user the possible options for language. If the user chooses a different language
+            // than the default, then the translation middleware will pick it up from the user state and
+            // translate messages both ways, i.e. user to bot and bot to user.
+            Activity reply = MessageFactory.text("Choose your language:");
+            CardAction esAction = new CardAction() {
+                {
+                    setTitle("Español");
+                    setType(ActionTypes.POST_BACK);
+                    setValue(ENGLISH_SPANISH);
+                }
+            };
+            CardAction enAction = new CardAction() {
+                {
+                    setTitle("English");
+                    setType(ActionTypes.POST_BACK);
+                    setValue(ENGLISH_ENGLISH);
+                }
+            };
+            List<CardAction> actions = new ArrayList<>(Arrays.asList(esAction, enAction));
+            SuggestedActions suggestedActions = new SuggestedActions();
+            suggestedActions.setActions(actions);
+            return turnContext.sendActivity(reply).thenApply(resourceResponse -> null);
+        }
+    }
+
+    private static CompletableFuture<Void> sendWelcomeMessage(TurnContext turnContext) {
+        for (ChannelAccount member: turnContext.getActivity().getMembersAdded()) {
+            // Greet anyone that was not the target (recipient) of this message.
+            // To learn more about Adaptive Cards, see https://aka.ms/msbot-adaptivecards for more details.
+            if (member.getId() != turnContext.getActivity().getRecipient().getId()) {
+                Attachment welcomeCard = createAdaptiveCardAttachment();
+                Activity response = MessageFactory.attachment(welcomeCard);
+                return turnContext.sendActivity(response)
+                    .thenCompose(task -> turnContext.sendActivity(MessageFactory.text(WELCOME_TEXT.toString()))
```

Comment:
This is weird, by `WELCOME_TEXT.toString()`? I know that is StringBuilder, but it should be parsed to String in the prop, not here